File: python_solutions/660.remove-9.py
from collections import Counter, defaultdict, OrderedDict, deque
from bisect import bisect_left, bisect_right
from functools import reduce, lru_cache
from typing import List
import itertools
import math
import heapq
import string
true = True
false = False
MIN, MAX, MOD = -0x3f3f3f3f, 0x3f3f3f3f, 1000000007
from collections import Counter, defaultdict, OrderedDict, deque
from bisect import bisect_left, bisect_right
from functools import reduce, lru_cache
from typing import List
import itertools
import math
import heapq
import string
true = True
false = False
MIN, MAX, MOD = -0x3f3f3f3f, 0x3f3f3f3f, 1000000007


# https://leetcode.com/problems/remove-9/description/
# Hard
class Solution:
    def newInteger(self, n: int) -> int:
        res, base = 0, 1
        while n > 0:
            res += n % 9 * base
            n //= 9
            base *= 10
        return res


print(Solution().newInteger(9))

File: python_solutions/660.remove-9.cpp
// Containers
#include <deque>
#include <iostream>
#include <map>
#include <numeric>
#include <queue>
#include <set>
#include <stack>
#include <unordered_map>
#include <unordered_set>
#include <vector>

#include "conf.d/say.h"

using namespace std;
int dirs[5] = {-1, 0, 1, 0, -1};

// --------------------------------------------------------
class Solution {
public:
  int newInteger(int n) {
    int64_t res = 0, base = 1;
    while (n != 0) {
      res += n % 9 * base;
      n /= 9;
      base *= 10;
    }
    return res;
  }
};

int main() {
  Solution s;
  say(s.newInteger(800000000));
}